packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx

```diff
@@ -59,14 +78,55 @@ export const fiatExchangesOptionsScreenOptions = ({
 }
 
 function FiatExchangeOptions({ route, navigation }: Props) {
+  const [providerLocalAmount, setProviderLocalAmount] = React.useState<BigNumber>()
+  const [providerlocalCurrency, setProviderLocalCurrency] = React.useState<LocalCurrencyCode>()
+  // All currency exchange calculations handled here as
+  // different providers may provide different fiat, CELO and cUSD rates
+
+  const localCurrency = useSelector(getLocalCurrencyCode)
   const { amount } = route.params
+
   const goToProvider = (screen: keyof StackParamList) => {
-    return () => navigation.navigate(screen, { amount })
+    return () =>
+      navigation.navigate(screen, {
+        localAmount: features.CUSD_MOONPAY_ENABLED
+          ? amount
+          : providerLocalAmount || new BigNumber(0),
+        currencyCode: providerlocalCurrency || FALLBACK_CURRENCY,
+      })
   }
   const onDismiss = () => {
     navigation.setParams({ isExplanationOpen: false })
   }
 
+  const dispatch = useDispatch()
+  React.useEffect(() => {
+    async function getMoonpayRates() {
+      const moonpayRates = await (await fetch(MOONPAY_RATE_API)).json()
+      const localMoonpayRate = moonpayRates[localCurrency]
+
+      if (localMoonpayRate) {
+        const localMoonpayAmount = amount.times(localMoonpayRate)
+        setProviderLocalCurrency(localCurrency)
+        setProviderLocalAmount(localMoonpayAmount)
+      } else if (moonpayRates[FALLBACK_CURRENCY]) {
+        // Local currency not available, use fallback currency
+        setProviderLocalCurrency(FALLBACK_CURRENCY)
+        setProviderLocalAmount(moonpayRates[FALLBACK_CURRENCY])
+      } else {
+        throw new Error('Unable to fetch Moonpay rates')
+      }
+    }
+    // TODO: Get rates from other providers when they are added
+    getMoonpayRates().catch((error) => {
+      Logger.error(
+        TAG,
+        `Failed to fetch Moonpay rate for ${localCurrency} at ${amount}, error: ${error}`
```

Comment:
pass the error in as the 3rd param to Logger.error instead of interpolating it into the message.

Reply:
👌